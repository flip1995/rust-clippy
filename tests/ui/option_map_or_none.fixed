// run-rustfix

#![allow(clippy::bind_instead_of_map)]

fn main() {
    let opt = Some(1);
    let bar = |_| {
        Some(1)
    };

    // Check `OPTION_MAP_OR_NONE`.
    // Single line case.
    let _ :Option<i32> = opt.map(|x| x + 1);
    // Multi-line case.
    #[rustfmt::skip]
    let _ :Option<i32> = opt.map(|x| x + 1);
    // function returning `Option`
    let _ :Option<i32> = opt.and_then(bar);
}
