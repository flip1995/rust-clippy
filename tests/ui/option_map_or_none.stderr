error: called `map_or(None, ..)` on an `Option` value. This can be done more directly by calling `map(..)` instead
  --> $DIR/option_map_or_none.rs:13:26
   |
LL |     let _ :Option<i32> = opt.map_or(None, |x| Some(x + 1));
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using `map` instead: `opt.map(|x| x + 1)`
   |
   = note: `-D clippy::option-map-or-none` implied by `-D warnings`

error: called `map_or(None, ..)` on an `Option` value. This can be done more directly by calling `map(..)` instead
  --> $DIR/option_map_or_none.rs:16:26
   |
LL |       let _ :Option<i32> = opt.map_or(None, |x| {
   |  __________________________^
LL | |                         Some(x + 1)
LL | |                        });
   | |_________________________^ help: try using `map` instead: `opt.map(|x| x + 1)`

error: called `map_or(None, ..)` on an `Option` value. This can be done more directly by calling `and_then(..)` instead
  --> $DIR/option_map_or_none.rs:20:26
   |
LL |     let _ :Option<i32> = opt.map_or(None, bar);
   |                          ^^^^^^^^^^^^^^^^^^^^^ help: try using `and_then` instead: `opt.and_then(bar)`

error: aborting due to 3 previous errors

